ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/tracing/flat/FlatTraceGenerator.java

```diff
@@ -530,11 +530,19 @@ private static FlatTrace.Context handleCallDataLoad(
 
   private static boolean hasRevertInSubCall(
       final TransactionTrace transactionTrace, final TraceFrame callFrame) {
-    return transactionTrace.getTraceFrames().stream()
-        .filter(traceFrame -> !traceFrame.equals(callFrame))
-        .takeWhile(traceFrame -> !traceFrame.getOpcode().equals("RETURN"))
-        .filter(traceFrame -> traceFrame.getOpcode().equals("REVERT"))
-        .anyMatch(traceFrame -> traceFrame.getDepth() == callFrame.getDepth());
+    for (int i = 0; i < transactionTrace.getTraceFrames().size(); i++) {
+      if (i + 1 < transactionTrace.getTraceFrames().size()) {
+        final TraceFrame next = transactionTrace.getTraceFrames().get(i + 1);
+        if (next.getDepth() == callFrame.getDepth()) {
+          if (next.getOpcode().equals("REVERT")) {
```

Comment:
We should add opcodeNumber to the TraceFrame and check on that instead of checking for string equality.